### PR TITLE
add new perf test for int8/bfloat16 dot product

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review
@havardpe FYI

### CODE REVIEW

File: tests/performance/tensor_eval/tensor_unstable_dot_product.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    results = []
+    results.append(get_first_result('qry64doc64',     'qry64'))
+    results.append(get_first_result('qry32doc32',     'qry32'))
+    results.append(get_first_result('qry16doc16',     'qry16'))
+    results.append(get_first_result('qry32doc16',     'qry32'))
+    results.append(get_first_result('qry32doc16cast', 'qry32'))
+    results.append(get_first_result('qry8doc8',       'qry8'))
+    results.append(get_first_result('qry32doc8',      'qry32'))
+    results.append(get_first_result('qry32doc8cast',  'qry32'))
+    # puts "RESULTS :::"
```
Comment (geirst) [inline tests/performance/tensor_eval/tensor_unstable_dot_product.rb:81]: Consider removing remains of debug logging.

File: tests/performance/tensor_eval/tensor_unstable_dot_product.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    # results.each { |r| puts "RESULT: #{r}" }
+    # puts "::: RESULTS"
+    results.each { |r| assert_equal(expect, r) }
+  end
+
+  def get_first_result(rank_profile, qf)
+    query_file = dirs.tmpdir + 'qf.' + qf
+    line = File.open(query_file) { |f| f.readline }
+    q = "#{line.chomp}&ranking=#{rank_profile}&timeout=10&hits=3&format=json"
+    #puts "run query: #{q}"
```
Comment (geirst) [inline tests/performance/tensor_eval/tensor_unstable_dot_product.rb:91]: Consider removing remains of debug logging.

**Review by geirst — APPROVED** [review]
👍

**Comment by havardpe** [comment]
should we have int8/int8 and bfloat16/bfloat16 with cast as well?

**Pull request merged**